core/CPUTopology.cpp

```diff
@@ -2,377 +2,204 @@
-            sparta::TreeNode::GROUP_IDX_NONE,
-            &factories->rename_rf
-        },
-        {
-            "dispatch",
-            "cpu.core*",
-            "Dispatch Unit",
-            sparta::TreeNode::GROUP_NAME_NONE,
-            sparta::TreeNode::GROUP_IDX_NONE,
-            &factories->dispatch_rf
-        },
-        {
-            "execute",
-            "cpu.core*",
-            "Execution Pipes",
-            "execute",
-            0,
-            &factories->execute_rf
-        },
-        {
-            "dcache",
-            "cpu.core*",
-            "Data Cache Unit",
-            sparta::TreeNode::GROUP_NAME_NONE,
-            sparta::TreeNode::GROUP_IDX_NONE,
-            &factories->dcache_rf
-        },
-        {
-            "mmu",
-            "cpu.core*",
-            "MMU Unit",
-            sparta::TreeNode::GROUP_NAME_NONE,
-            sparta::TreeNode::GROUP_IDX_NONE,
-            &factories->mmu_rf
-        },
-        {
-            "tlb",
-            "cpu.core*.mmu",
-            "TLB Unit",
-            sparta::TreeNode::GROUP_NAME_NONE,
-            sparta::TreeNode::GROUP_IDX_NONE,
-            &factories->tlb_rf,
-            true
-        },
-        {
-            "lsu",
-            "cpu.core*",
-            "Load-Store Unit",
-            sparta::TreeNode::GROUP_NAME_NONE,
-            sparta::TreeNode::GROUP_IDX_NONE,
-            &factories->lsu_rf
-        },
-        {
-            "l2cache",
-            "cpu.core*",
-            "L2Cache Unit",
-            sparta::TreeNode::GROUP_NAME_NONE,
-            sparta::TreeNode::GROUP_IDX_NONE,
-            &factories->l2cache_rf
-        },
-        {
-            "biu",
-            "cpu.core*",
-            "Bus Interface Unit",
-            sparta::TreeNode::GROUP_NAME_NONE,
-            sparta::TreeNode::GROUP_IDX_NONE,
-            &factories->biu_rf
-        },
-        {
-            "mss",
-            "cpu.core*",
-            "Memory Sub-System",
-            sparta::TreeNode::GROUP_NAME_NONE,
-            sparta::TreeNode::GROUP_IDX_NONE,
-            &factories->mss_rf
-        },
-        {
-            "rob",
-            "cpu.core*",
-            "ROB Unit",
-            sparta::TreeNode::GROUP_NAME_NONE,
-            sparta::TreeNode::GROUP_IDX_NONE,
-            &factories->rob_rf
-        },
-        {
-            "preloader",
-            "cpu.core*",
-            "Preloader Facility",
-            sparta::TreeNode::GROUP_NAME_NONE,
-            sparta::TreeNode::GROUP_IDX_NONE,
-            &factories->preloader_rf
-        },
-        {
-            "mavis",
-            "cpu.core*",  // Each core can have its own decoder
-            "Mavis Decoding Functional Unit",
-            sparta::TreeNode::GROUP_NAME_NONE,
-            sparta::TreeNode::GROUP_IDX_NONE,
-            &factories->mavis_rf
-        }
-    };
+    units = {{"core*", "cpu", "Core *", sparta::TreeNode::GROUP_NAME_NONE,
```

Comment:
Can you undo the formatting on this?

Reply:
Fixed in [19ed4b7](https://github.com/riscv-software-src/riscv-perf-model/pull/140/commits/19ed4b7911a0f115a2f9e2c2c19e5ad3324bbede)